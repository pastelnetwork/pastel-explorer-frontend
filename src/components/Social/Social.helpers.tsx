import * as URLS from '@utils/constants/urls';
import TelegramIcon from '@components/SvgIcon/TelegramIcon';
import TwitterIcon from '@components/SvgIcon/TwitterIcon';
import DiscordIcon from '@components/SvgIcon/DiscordIcon';
import Medium from '@components/SvgIcon/Medium';

export const footerIcons = [
  { id: 2, url: URLS.TELEGRAM_URL, icon: <TelegramIcon /> },
  { id: 1, url: URLS.TWITTER_URL, icon: <TwitterIcon /> },
  { id: 3, url: URLS.DISCORD_URL, icon: <DiscordIcon /> },
  { id: 4, url: URLS.MEDIUM_URL, icon: <Medium /> },
];
