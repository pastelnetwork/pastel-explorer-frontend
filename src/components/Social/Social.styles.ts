import styled from 'styled-components';
import { IconButton } from '@mui/material';

export const IconButtonLink = styled(IconButton)<{
  target: string;
  href: string;
  className: string;
}>`
  width: 30px;
  height: 30px;
  padding: 6px;
  border-radius: 100%;
  border: 1px solid ${props => props.theme.sidebar.menu.border};
  transition: all 0.5s ease;

  &:hover {
    background: none;
  }

  svg {
    max-width: 100%;
    fill: ${props => props.theme.sidebar.menu.border};
    transition: all 0.5s ease;
  }

  &:hover {
    border-color: ${props => props.theme.sidebar.menu.active};

    svg {
      fill: ${props => props.theme.sidebar.menu.active};
    }
  }
`;

export const Items = styled.ul`
  display: flex;
  list-style: none;
  margin-top: 25px;

  ${props => props.theme.breakpoints.down('xs')} {
    justify-content: center;
    margin: 20px 0 0;
    padding: 0;
  }
`;

export const Item = styled.li`
  margin-right: 15px;

  &:last-child {
    margin-right: 0;
  }
`;
