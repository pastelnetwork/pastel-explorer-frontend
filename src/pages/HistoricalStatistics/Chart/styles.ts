import { makeStyles } from '@material-ui/core/styles';
import { TAppTheme } from '@theme/index';

export const eChartLineStyles = makeStyles((theme: TAppTheme) => ({
  container: {
    margin: '0 auto',
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    height: '100%',
    width: '100%',
    padding: '0 20px',
    textAlign: 'center',
    boxSizing: 'border-box',
    position: 'relative',
    color: 'rgb(171, 170, 193)',
    [theme.breakpoints.down('sm')]: {
      padding: 0,
    },
  },
  lineChartWrap: {
    width: '100%',
    height: 'auto',
    marginBottom: 20,
  },
  lineChartContainer: {
    display: 'block',
    width: '100%',
    height: '100%',
    backgroundColor: '#212121',
  },
  lineChartTitle: {
    width: '100%',
    textAlign: 'left',
    fontSize: '30px',
    [theme.breakpoints.down('sm')]: {
      fontSize: 15,
    },
  },
  lineChartHeader: {
    display: 'flex',
    width: '100%',
    justifyContent: 'space-between',
    paddingTop: '30px',
    paddingBottom: '10px',
    [theme.breakpoints.down('sm')]: {
      paddingTop: 10,
      paddingLeft: 10,
      justifyContent: 'flex-start',
      flexDirection: 'column',
    },
  },
  uploadButton: {
    display: 'flex',
    alignItems: 'center',
    backgroundColor: '#6151d0',
    color: 'white',
    textAlign: 'center',
    fontWeight: 'bold',
    fontSize: '14px',
    padding: '8px 16px',
    borderRadius: '4px',
    marginLeft: '8px',
    marginRight: '8px',
    '&:hover': {
      cursor: 'pointer',
      opacity: 0.8,
    },
  },
  reactECharts: {
    width: '100%',
    height: 'calc(100vh - 450px) !important',
  },
  granularitySelect: {
    display: 'flex',
    width: '100%',
    maxWidth: 200,
    justifyContent: 'space-around',
    alignItems: 'center',
    marginTop: 15,
    marginRight: 50,
  },
  periodSelect: {
    display: 'flex',
    minWidth: '200px',
    // justifyContent: 'space-around',
    alignItems: 'center',
    marginTop: '15px',
    [theme.breakpoints.down('sm')]: {
      marginTop: 5,
    },
  },
  lineChartDownloadButtonBar: {
    display: 'flex',
    width: '100%',
    justifyContent: 'flex-end',
    height: '30px',
    [theme.breakpoints.down('sm')]: {
      marginTop: 10,
      justifyContent: 'flex-start',
    },
  },
  lineChartFooter: {
    display: 'flex',
    width: '100%',
    justifyContent: 'space-around',
    alignItems: 'center',
    paddingBottom: 30,
    [theme.breakpoints.down('sm')]: {
      flexWrap: 'wrap',
      justifyContent: 'flex-start',
      paddingBottom: 10,
    },
  },
  lineChartThemeSelect: {
    display: 'flex',
    minWidth: '150px',
    justifyContent: 'space-around',
    alignItems: 'center',
  },
  themeSelectButton: {
    width: '30px',
    height: '30px',
    borderRadius: '5px',
  },
  filterButton: {
    backgroundColor: '#eee',
    color: '#212124',
    textAlign: 'center',
    fontWeight: 'bold',
    fontSize: '12px',
    padding: '4px 8px',
    borderRadius: '3px',
    borderWidth: '1px',
    borderColor: '#212124',
    marginLeft: '4px',
    marginRight: '4px',
    '&:hover': {
      cursor: 'pointer',
      opacity: '0.8',
    },
  },
  activeButton: {
    backgroundColor: '#c3921f',
    borderColor: '#c3921f',
    color: '#fff',
  },
  activeThemeButton: {
    borderColor: '#008000',
    borderWidth: '5px',
  },
}));
