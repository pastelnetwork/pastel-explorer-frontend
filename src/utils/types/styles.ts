/* eslint-disable @typescript-eslint/no-explicit-any */
/* eslint-disable @typescript-eslint/ban-types */
import { SpacingProps } from '@material-ui/system';
import { Theme } from '@material-ui/core';

export type GlobalStyleProps = {
  theme: Theme & { palette: any };
};

export interface MuiButtonSpacingType extends SpacingProps {
  component?: React.PropsWithoutRef<{}>;
  to?: string;
}
