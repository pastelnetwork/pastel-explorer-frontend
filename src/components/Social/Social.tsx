import React from 'react';

import { footerIcons } from './Social.helpers';
import * as Styles from './Social.styles';

interface ISocial {
  className?: string;
}

export default function Social({ className = '' }: ISocial): JSX.Element {
  return (
    <Styles.Items className={className}>
      {footerIcons.map(({ id, url, icon }) => (
        <Styles.Item key={id}>
          <Styles.IconButtonLink target="_blank" href={url} className="social-icon">
            {icon}
          </Styles.IconButtonLink>
        </Styles.Item>
      ))}
    </Styles.Items>
  );
}

Social.defaultProps = {
  className: '',
};
