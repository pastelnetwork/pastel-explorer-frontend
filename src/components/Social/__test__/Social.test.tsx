import { shallow } from 'enzyme';

import 'jest-styled-components';

import Social from '../Social';
import * as Styles from '../Social.styles';

describe('components/Social', () => {
  const wrapper = shallow(<Social />);

  test('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should render <Styles.Items>', () => {
    expect(wrapper.find(Styles.Items).length).toBeGreaterThanOrEqual(1);
  });

  test('should render <Styles.Item>', () => {
    expect(wrapper.find(Styles.Item).length).toBeGreaterThanOrEqual(1);
  });

  test('should render <Styles.IconButtonLink>', () => {
    expect(wrapper.find(Styles.IconButtonLink).length).toBeGreaterThanOrEqual(1);
  });
});
