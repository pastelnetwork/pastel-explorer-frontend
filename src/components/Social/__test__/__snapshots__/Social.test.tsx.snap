// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Social renders correctly 1`] = `ShallowWrapper {}`;
